lint: unnamedResult panics on GOROOT

Execute `gocritic check-project -enable=unnamedResult $GOROOT/src`

```
check-package: unnamedResult: error: runtime error: index out of range
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 2628 [running]:
github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile.func1.1(0xc02291a6d0, 0xc0225befc0)
	GOPATH/src/github.com/go-critic/go-critic/cmd/criticize/main.go:184 +0x138
panic(0x7b0a60, 0xb07990)
	GOROOT/src/runtime/panic.go:494 +0x1b9
github.com/go-critic/go-critic/lint.(*unnamedResultChecker).CheckFuncDecl(0xc04e8c0718, 0xc0170f9b00)
	GOPATH/src/github.com/go-critic/go-critic/lint/unnamedResult_checker.go:30 +0x16b
github.com/go-critic/go-critic/lint.wrapFuncDeclChecker.func1(0xc0171a1100)
	GOPATH/src/github.com/go-critic/go-critic/lint/wrappers.go:62 +0x92
github.com/go-critic/go-critic/lint.(*Checker).Check(0xc0225befc0, 0xc0171a1100, 0xc02291a6d0, 0xc0225befc0, 0x0)
	GOPATH/src/github.com/go-critic/go-critic/lint/lint.go:94 +0x3c
github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile.func1(0xc02291a6d0, 0xc0171a1100, 0xc0001562a0, 0xc0225befc0)
	GOPATH/src/github.com/go-critic/go-critic/cmd/criticize/main.go:192 +0x94
created by github.com/go-critic/go-critic/cmd/criticize.(*linter).checkFile
	GOPATH/src/github.com/go-critic/go-critic/cmd/criticize/main.go:174 +0xac
check-project: lint error: exit status 2
```
Related code:
https://github.com/go-critic/go-critic/blob/447365f53e9dd16963a1caa93e2f8365df1f5ff6/lint/unnamedResult_checker.go#L28-L32

Can be reproduced with this negative test:
```go
func (f foo) g10() (x, y int) {
	return x, y
}
```

--- a/cmd/gocritic/testdata/unnamedResult/negative_tests.go
+++ b/cmd/gocritic/testdata/unnamedResult/negative_tests.go
@@ -35,3 +35,15 @@ func (f foo) g8() (x int) {
 func (f foo) g9() (x, y int, z float64) {
 	return 0, 0, 0
 }
+
+func (f foo) g10() (x, y int) {
+	return 0, 0
+}
+
+func g11() (x, y, z int) {
+	return 0, 0, 0
+}
+
+func g12() (x, y int, _, w float64) {
+	return 0, 0, 0, 0
+}

--- a/cmd/gocritic/testdata/unnamedResult/positive_tests.go
+++ b/cmd/gocritic/testdata/unnamedResult/positive_tests.go
@@ -46,3 +46,8 @@ func (f *foo) f8() (bool, bool) {
 func (f *foo) f9() (bool, func() int) {
 	return false, nil
 }
+
+/// consider to give name to results
+func f10() (int, int, float64, float64) {
+	return 0, 0, 0, 0
+}

--- a/lint/unnamedResult_checker.go
+++ b/lint/unnamedResult_checker.go
@@ -26,8 +26,9 @@ func (c *unnamedResultChecker) CheckFuncDecl(decl *ast.FuncDecl) {
 		return
 
 	case results.NumFields() == 2:
-		if !c.isBoolOrError(results.List[0].Type) &&
-			c.isBoolOrError(results.List[1].Type) {
+		typ1, typ2 := c.getResultsTypes(results.List)
+		if len(results.List[0].Names) == 2 ||
+			(!c.isBoolOrError(typ1) && c.isBoolOrError(typ2)) {
 			// no need to name results for (T, error) or (T, bool)
 		} else {
 			c.warn(decl)
@@ -45,6 +46,13 @@ func (c *unnamedResultChecker) CheckFuncDecl(decl *ast.FuncDecl) {
 
 func (c *unnamedResultChecker) warn(n ast.Node) {
 	c.ctx.Warn(n, "consider to give name to results")
+}
+
+func (c *unnamedResultChecker) getResultsTypes(fields []*ast.Field) (res1, res2 ast.Expr) {
+	if len(fields) == 2 {
+		return fields[0].Type, fields[1].Type
+	}
+	return fields[0].Type, fields[0].Type
 }
 
 func (c *unnamedResultChecker) isBoolOrError(expr ast.Expr) bool {

